### PR TITLE
separate routines into separate files so predictor and corrector aren…

### PR DESCRIPTION
**Pull request by asalmgren**
…'t so long

### CODE REVIEW

File: src/incflo_update_density.cpp [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+                    });
+                }
+            } // mfi
+
+            if (step_type == StepType::Predictor) {
+                // Fill ghost cells of the new density field so that we can define density_nph
+                //      on the valid region grown by 1
+                fillpatch_density(lev, m_t_new[lev], ld.density, ng);
+
+                for (MFIter mfi(ld.velocity,TilingIfNotGPU()); mfi.isValid(); ++mfi)
```
Comment (WeiqunZhang) [inline src/incflo_update_density.cpp]: We can add omp parallel here. Or just call
```
MultiFab::LinComb(ld.density_nph,
                  Real(0.5), ld.density, 0,
                  Real(0.5), ld.density_o, 0,
                  0, 1, 1);
```

File: src/incflo_apply_corrector.cpp [hunk 2]
```diff
@@ -156,325 +143,26 @@ void incflo::ApplyCorrector()
                        get_density_new_const(), GetVecOfConstPtrs(vel_eta));
     }
 
-    if (m_advect_tracer && m_diff_type == DiffusionType::Explicit) {
-        compute_laps(get_laps_new(), get_tracer_new_const(),
```
Comment (WeiqunZhang) [inline src/incflo_apply_corrector.cpp:160]: My understanding is that this has been moved to `update_tracer`, which is called after `update_density` updates the new density. So the density used by compute_laps has changed in this PR. Is that intentional?

Comment (asalmgren) [reply src/incflo_apply_corrector.cpp:160]: density isn't actually used in compute_laps -- I'll remove that from the argument list.

File: src/incflo_update_density.cpp (hunk 1 above)
Comment (asalmgren) [reply src/incflo_update_density.cpp]: I'll commit this change

**Review by WeiqunZhang — APPROVED** [review]

**Pull request merged**